lib/color/color.go

```diff
@@ -143,3 +145,204 @@ const (
 	Empty = ""
 	None  = "none"
 )
+
+type RGB struct {
+	Red   uint8
+	Green uint8
+	Blue  uint8
+}
+
+// https://github.com/go-playground/colors/blob/main/rgb.go#L89
+func (c *RGB) IsLight() bool {
+	r := float64(c.Red)
+	g := float64(c.Green)
+	b := float64(c.Blue)
+
+	hsp := math.Sqrt(0.299*math.Pow(r, 2) + 0.587*math.Pow(g, 2) + 0.114*math.Pow(b, 2))
+
+	return hsp > 130
+}
+
+// https://gist.github.com/CraigChilds94/6514edbc6a2db5e434a245487c525c75
+func Hex2RGB(hex string) (RGB, error) {
+	var rgb RGB
+	if hex[0:1] == "#" {
+		hex = hex[1:]
+	} else {
+		return RGB{}, fmt.Errorf("cannot parse hex color %v", hex)
+	}
+	values, err := strconv.ParseUint(hex, 16, 32)
+	if err != nil {
+		return RGB{}, err
+	}
+
+	rgb = RGB{
+		Red:   uint8(values >> 16),
+		Green: uint8((values >> 8) & 0xFF),
+		Blue:  uint8(values & 0xFF),
+	}
+
+	return rgb, nil
+}
+
+// https://www.w3schools.com/cssref/css_colors.php
```

Comment:
w3schools is ancient. use a more recent source. e.g. https://147colors.com/rebeccapurple/

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/named-color is probably best

Reply:
Changed it to w3 official source

Reply:
nice that's even better.